snips/customtts.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+#set -x
+API_KEY=$HASSIO_TOKEN
+PLATFORM=$1
+FILE=$2
+LANG=$3
+TEXT=$4
+
+MESSAGE="\"{\\\"message\\\": \\\"$TEXT\\\", \\\"platform\\\": \\\"$PLATFORM\\\"}\""
+echo "$MESSAGE"
+
+RESPONSE=$(eval curl -s -H \"x-ha-access: "$API_KEY"\" -H \"Type: application/json\" http://hassio/homeassistant/api/tts_get_url -d "$MESSAGE")
+if [ "$RESPONSE" = "" ]; then
+    exit 1
+fi
+echo "$RESPONSE"
+
+URL=$(echo "$RESPONSE" | jq --raw-output '.url')
+if [ "$URL" = "" ]; then
+    exit 1
+fi
+
+rm /tmp/temp.mp3
+curl -s -H "x-ha-access: $API_KEY" "$URL" -o /tmp/temp.mp3
+if [ -f /tmp/temp.mp3 ]; then
+  /usr/bin/mpg123 -w "$FILE" /tmp/temp.mp3
+fi
+rm /tmp/temp.mp3
```

Comment:
Maybe we can directly play this with need to download the file?

Reply:
The snips server only accepts WAV audio. We could play it but then it doesn't go through the snips dialog manager so can get odd behaviour like it hearing itself speak.

Reply:
To clarify, google and amazon both support PCM but it would still need to be massaged into WAV files via sox or some other program, so best to just stick with mp3 and convert it here.

Reply:
I mean, can we play the URL on mpg123 ?

Reply:
No. It wouldn't be done through snips dialog manager, so snips wouldn't know if it was playing so wouldn't know when it was done and could be listening at the same time (ie hearing itself talk) Also if you had multiroom installations of snips it wouldn't play on the correct site.